### PR TITLE
Sort goals by name in UI

### LINKED ISSUE #17950: Sort goals alphabetically in "Manage Goals" and other places where Goals are listed
## Proposed acceptance criteria

Sort goals alphabetically everywhere they are listed in a list including:

* in Manage Goals
*  in the "Goals Overview" report page (showing summary of each goal)
*  in the "Goals" selector (under Goals left menu when more than X goals created)

Note: in Multi attribution "Goals" selector, Goals are already sorted alphabetically.

## Original issue

The goals ("Manage goals") are sorted numerically by IDs. From a User's
point of view, it would be useful to be able to sort the goals manually
or at least alphabetically.

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

Goals are currently sorted by their ID all across Matomo. This makes it hard to find a specific goal in the UI, as the IDs aren't visible (besides in Management).
To make the UI more use friendly this PR changes the sort of Goals in UI only. API responses will be still sorted by id.

Changed places:
* Goal management
* Goal listing in menu
* Goal sparklines in goal overview
* Goal listing in goal report table

fixes #17950

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/Goals/API.php [hunk 1]
```diff
@@ -80,10 +80,11 @@ public function getGoal($idSite, $idGoal)
      * Returns all Goals for a given website, or list of websites
      *
      * @param string|array $idSite Array or Comma separated list of website IDs to request the goals for
+     * @param bool $orderByName
      *
      * @return array Array of Goal attributes
      */
-    public function getGoals($idSite)
+    public function getGoals($idSite, bool $orderByName = false): array
```
Comment (caddoo) [inline plugins/Goals/API.php:87]: Is there anything you could do to clean up this function/method a bit while you are working in this area?

Comment (sgiehl) [reply plugins/Goals/API.php:87]: What should be cleaned up in this function? It's actually quite straight forward. As the method is used very often it has a non persistent caching

Comment (michalkleiner) [reply plugins/Goals/API.php:87]: I also had a look and I wasn't sure myself what I'd improve in the method.

Comment (caddoo) [reply plugins/Goals/API.php:87]: Well just the method is doing many things like:
- Getting goals, while also setting (in cache)
- Cleaning goals
- Sorting goals

Breaking it into smaller methods would be easier to read, and then if that looks messy having all that logic in the `API` it should be elsewhere.

You don't need to change anything, especially if it's just my preference, just wanted to raise it.

Comment (sgiehl) [reply plugins/Goals/API.php:87]: Normally I would have put the sorting into the model, but in this special case that would have meant we would need to cache twice - one sorted and one unsorted. So in terms of performance I decided to do the sorting only when needed on the already cached result.

**Review by mneudert — APPROVED** [review]

**Pull request merged**